target/ppc/translate/vsx-impl.c.inc

```diff
@@ -2124,6 +2124,88 @@ static void gen_xvxsigdp(DisasContext *ctx)
     tcg_temp_free_i64(xbl);
 }
 
+static bool do_lstxv(DisasContext *ctx, arg_D *a, bool store, bool paired)
+{
+    TCGv ea;
+    TCGv_i64 xt;
+    MemOp mop;
+    int offset;
+
+    if (paired) {
+        REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    }
+
+    REQUIRE_VSX(ctx);
+
+    ea = tcg_temp_new();
+
+    mop = DEF_MEMOP(MO_Q);
+
+    gen_set_access_type(ctx, ACCESS_INT);
+
+    xt = tcg_temp_new_i64();
+
+    if (ctx->le_mode) {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si + (paired ? 24 : 8)), ea);
+        offset = -8;
+    } else {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si), ea);
+        offset = 8;
+    }
+
+    if(store) {
+        get_cpu_vsrh(xt, a->rt);
+        tcg_gen_qemu_st_i64(xt, ea, ctx->mem_idx, mop);
+        gen_addr_add(ctx, ea, ea, offset);
+        get_cpu_vsrl(xt, a->rt);
+        tcg_gen_qemu_st_i64(xt, ea, ctx->mem_idx, mop);
+        if (paired) {
+            gen_addr_add(ctx, ea, ea, offset);
+            get_cpu_vsrh(xt, a->rt + 1);
+            tcg_gen_qemu_st_i64(xt, ea, ctx->mem_idx, mop);
+            gen_addr_add(ctx, ea, ea, offset);
+            get_cpu_vsrl(xt, a->rt + 1);
+            tcg_gen_qemu_st_i64(xt, ea, ctx->mem_idx, mop);
+        }
+    } else {
+        tcg_gen_qemu_ld_i64(xt, ea, ctx->mem_idx, mop);
+        set_cpu_vsrh(a->rt, xt);
+        gen_addr_add(ctx, ea, ea, offset);
+        tcg_gen_qemu_ld_i64(xt, ea, ctx->mem_idx, mop);
+        set_cpu_vsrl(a->rt, xt);
+        if (paired) {
+            gen_addr_add(ctx, ea, ea, offset);
+            tcg_gen_qemu_ld_i64(xt, ea, ctx->mem_idx, mop);
+            set_cpu_vsrh(a->rt + 1, xt);
+            gen_addr_add(ctx, ea, ea, offset);
+            tcg_gen_qemu_ld_i64(xt, ea, ctx->mem_idx, mop);
+            set_cpu_vsrl(a->rt + 1, xt);
+        }
+    }
+
+    tcg_temp_free(ea);
+    tcg_temp_free_i64(xt);
+    return true;
+}
+
+static bool do_lstxv_PLS_D(DisasContext *ctx, arg_PLXV *a, bool store, bool paired)
+{
+    arg_D d;
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    if (unlikely(!resolve_PLS_D(ctx, &d, a))) {
```

Comment:
Not sure what's the best form, but we're not wrapping the other ``resolve_PLS_D()`` calls inside ``unlikely``. Just for consistency, I would drop the ``unlikely`` here.